src/DssSpell.sol

```diff
@@ -110,24 +110,43 @@ contract DssSpell {
     }
 
     modifier limited {
+        require(nextCastTime() == now, "DSSSpell/outside-office-hours");
+        _;
+    }
+
+    function nextCastTime() public view returns (uint256) {
+        require(eta != 0, "DSSSpell/spell-not-scheduled");
+        uint256 castTime = now > eta ? now : eta;
+
         if (officeHours) {
-            uint day = (now / 1 days + 3) % 7;
-            require(day < 5, "Can only be cast on a weekday");
-            uint hour = now / 1 hours % 24;
-            require(hour >= 14 && hour < 21, "Outside office hours");
+            uint256 day    = (castTime / 1 days + 3) % 7;
```

Comment:
Are we confortable placing this whole chunk of code in a spell without safe math?

Reply:
I didn't use safemath here because I figured that we were working with small enough numbers (timestamps) that overflow would never happen. @kmbarry1 @brianmcmichael whatre your thoughts?

Reply:
I don't see a need for safe math here, personally.